src/database/operations/create/createBulkUpload.ts

```diff
@@ -0,0 +1,21 @@
+import { db } from '../../db';
+import type { BulkUpload, InternallyWritableBulkUpload } from '../../../types';
+
+export const createBulkUpload = async (
+	createValues: InternallyWritableBulkUpload,
+): Promise<BulkUpload> => {
+	const { fileName, sourceKey, status } = createValues;
+
+	const result = await db.sql<BulkUpload>('bulkUploads.insertOne', {
+		fileName,
+		sourceKey,
+		status,
+	});
+	const object = result.rows[0];
+	if (object === undefined) {
+		throw new Error(
```

Comment:
This `throw` statement seems common enough across the `create` functions to warrant a dedicated class that extends `Error`.

Reply:
For sure -- I haven't made an issue for this but my plan is to actually refactor this ENTIRE repeated block into a "loadItem" utility that will handle the undefined check as well.

That said... after that refactor this is an unreachable error so I'm not sure we should make a custom type for it (maybe UnreachableError would be fun though!)

Let's decide at that PR though if that's OK :D